src/sfizz/ADSREnvelope.cpp

```diff
@@ -113,13 +116,24 @@ void ADSREnvelope::getBlockInternal(absl::Span<Float> output) noexcept
             while (count < size && delay-- > 0) {
                 currentValue = start;
                 output[count++] = currentValue;
+                attackCount = 0;
             }
             if (delay <= 0)
+            {
+                attackCount = 0;
                 currentState = State::Attack;
+            }
             break;
         case State::Attack:
-            while (count < size && (currentValue += attackStep) < 1)
+            while (count < size && (currentValue) < 1)
+            {
+                if (attackShape <= 0)
+                    currentValue = start + (1 - start) * pow(attackCount, -attackShape + 1);
```

Comment:
`pow` is pretty hardcore in such a tight loop, I think we could at least extract it before the while.

Reply:
Hmm. Well, since it appears the pow function doesn't make use of any loop-specific variables, at least for this line, critique accepted.